apps/site/assets/ts/schedule/components/schedule-finder/ScheduleFinderModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ReactElement, useState } from "react";
+import { DirectionId, Route } from "../../../__v3api";
+import {
+  SimpleStopMap,
+  RoutePatternsByDirection,
+  ServiceInSelector,
+  ScheduleNote as ScheduleNoteType,
+  SelectedOrigin
+} from "../__schedule";
+import Modal from "../../../components/Modal";
+import OriginModalContent from "./OriginModalContent";
+import ScheduleModalContent from "./ScheduleModalContent";
+
+export type Mode = "origin" | "schedule";
+
+interface Props {
+  closeModal: () => void;
+  directionChanged?: (direction: DirectionId) => void;
+  initialDirection: DirectionId;
+  initialMode: Mode;
+  initialOrigin: SelectedOrigin;
+  originChanged?: (origin: SelectedOrigin) => void;
+  route: Route;
+  routePatternsByDirection: RoutePatternsByDirection;
+  scheduleNote: ScheduleNoteType | null;
+  services: ServiceInSelector[];
+  stops: SimpleStopMap;
+  today: string;
+}
+
+export default ({
+  closeModal,
+  directionChanged,
+  initialDirection,
+  initialMode,
+  initialOrigin,
+  originChanged,
+  route,
+  routePatternsByDirection,
+  scheduleNote,
+  services,
+  stops,
+  today
+}: Props): ReactElement => {
+  const [direction, setDirection] = useState(initialDirection);
+  const [mode, setMode] = useState(initialMode);
+  const [origin, setOrigin] = useState(initialOrigin);
+
+  const handleChangeDirection = (newDirection: DirectionId): void => {
+    setDirection(newDirection);
+    setOrigin(null);
+    setMode("origin");
+    if (directionChanged) directionChanged(newDirection);
+    if (originChanged) originChanged(null);
+  };
+
+  const handleChangeOrigin = (newOrigin: SelectedOrigin): void => {
+    setOrigin(newOrigin);
+    setMode("schedule");
+    if (originChanged) originChanged(newOrigin);
+  };
+
+  return (
+    <Modal
+      focusElementId={mode === "origin" ? "origin-filter" : "modal-close"}
+      ariaLabel={{
+        label:
+          mode === "origin"
+            ? "Choose Origin Stop"
+            : `Schedules to ${route.direction_names[direction]}`
+      }}
+      className={mode === "origin" ? "schedule-finder__origin-modal" : ""}
+      closeModal={closeModal}
+    >
+      {() => (
```

Comment:
The existing `Modal` logic expects its `children` to be a function that returns a JSX element, rather than a JSX element. I'm not sure exactly why this is, but I'd prefer not to mess with it unless we know it's causing issues.

Reply:
...I may try just removing this aspect of it and seeing if everything still works, though, because this particular modal is the only user of `Modal` and doesn't seem to use the argument that is passed to the `children` function. I wonder if this is a leftover of a different approach to modals.

Reply:
Seems like this works fine. I've made this change now.